### PR TITLE
[#6717] Update storage tasks

### LINKED ISSUE #6717: Commit tweaks to storage.rb
We've currently got this diff in production. Let's check in these lines so that a deploy doesn't obliterate them.

```diff
diff --git a/lib/tasks/storage/storage.rb b/lib/tasks/storage/storage.rb
index c91fa45d6..62926f56d 100644
--- a/lib/tasks/storage/storage.rb
+++ b/lib/tasks/storage/storage.rb
@@ -26,6 +26,7 @@ class Storage

       erase_line
       print "#{prefix}: Migrated #{index + 1}/#{count}"
+    rescue Errno::ENOENT
     end

     erase_line
@@ -93,7 +94,7 @@ class Storage
   def unattached_files
     @klass.left_joins(:"#{@association}_attachment").where(
       active_storage_attachments: { id: nil }
-    )
+    ).joins(:incoming_message)
   end

   def blobs
```

### PR DESCRIPTION
**Pull request by gbp**
## Relevant issue(s)

Fixes #6717

## What does this do?

Update storage tasks for issues found when migrating WDTK to `ActiveStorage`

## Why was this needed?

- Reduce number of downloads from cloud storage for new raw email responses
- Help make re-users experience easier.

### CODE REVIEW

File: lib/tasks/storage/storage.rb [hunk 1]
```diff
@@ -109,7 +109,7 @@ def mirrorable_blobs
   end
 
   def promotable_blobs
-    mirrorable_blobs
+    mirrorable_blobs.where(created_at: (..7.days.ago))
```
Comment (garethrees) [inline lib/tasks/storage/storage.rb:119]: TIL! Assume this is shorthand for `Time.zone.now..7.days.ago`?

File: lib/tasks/storage/storage.rb [hunk 2]
```diff
@@ -26,6 +26,13 @@ def migrate
 
       erase_line
       print "#{prefix}: Migrated #{index + 1}/#{count}"
+
+    rescue Errno::ENOENT
+      erase_line
+      Kernel.silence_warnings do
+        puts "#{prefix};ID=#{file.id}: Missing #{file.filepath}."
```
Comment (garethrees) [inline lib/tasks/storage/storage.rb:33]: `$stderr.puts` since it's an error case?

**Review by garethrees — APPROVED** [review]
Haven't run, but LGTM.

File: lib/tasks/storage/storage.rb (hunk 1 above)
Comment (gbp) [reply lib/tasks/storage/storage.rb:119]: Available since https://github.com/mysociety/alaveteli/issues/5224

**Pull request merged**